src/sentry/event_manager.py

```diff
@@ -2455,6 +2459,52 @@ def _save_aggregate_performance(jobs: Sequence[PerformanceJob], projects: Projec
                 EventPerformanceProblem(event, performance_problems_by_hash[problem_hash]).save()
 
 
+@metrics.wraps("save_event.send_occurrence_to_platform")
+def _send_occurrence_to_platform(jobs: Sequence[Job], projects: ProjectsMapping) -> None:
+    for job in jobs:
+        event = job["event"]
+        project = event.project
+        event_id = event.event_id
+
+        use_platform = options.get("performance.issues.send_to_issues_platform", False)
+        if use_platform:
+            per_project_option = project.get_option(
+                "sentry:performance_issue_send_to_issues_platform", False
+            )
+            if per_project_option:
+                performance_problems = job["performance_problems"]
+                for problem in performance_problems:
+                    # handle only N+1 db query detector first
+                    if problem.type.type_id == PerformanceNPlusOneGroupType.type_id:
+                        occurrence = IssueOccurrence(
+                            id=uuid.uuid4().hex,
+                            resource_id=None,
+                            project_id=project.id,
+                            event_id=event_id,
+                            fingerprint=[problem.fingerprint],
+                            type=problem.type,
+                            issue_title=problem.title,
+                            subtitle=problem.desc,
+                            culprit=event.transaction,
+                            evidence_data=problem.evidence_data,
+                            evidence_display=problem.evidence_display,
+                            detection_time=event.datetime,
+                            level="info",
+                        )
+
+                        topic = settings.KAFKA_INGEST_OCCURRENCES
+                        cluster_name = settings.KAFKA_TOPICS[topic]["cluster"]
+                        cluster_options = get_kafka_producer_cluster_options(cluster_name)
+                        producer = Producer(cluster_options)
+
+                        producer.produce(
+                            topic=topic,
+                            key=None,
+                            value=json.dumps(occurrence.to_dict()),
+                        )
+                        producer.flush()
```

Comment:
I wasn't sure whether we were using this producer correctly for a high volume environment, so asked in sns. Turns out that we aren't using it correctly, and flushing on every message will slow things down a fair bit.

Slack thread is [here](https://sentry.slack.com/archives/CLTE78L73/p1678931612520379), but I'll summarise a bit: 

 - It's slightly better to cache the `Producer` and not instantiate it every time
 - Flushing on every message is bad, since it makes what was an async call into a sync one and will slow down processing
 - We should consider using the Arroyo `Producer` since it returns futures for each flush that we can wait on.
 - We'll need to keep the futures from each `produce` call around somewhere so that we can be sure they complete. Recommendations here are to store the futures in a global, then use `atexit` to await the futures.